// telegram-node-bot/lib/models/Models.js
module.exports = {
   Update: require('./Update'),
   User: require('./User'),
   Chat: require('./Chat'),
   Message: require('./Message'),
   MessageEntity: require('./MessageEntity'),
   PhotoSize: require('./PhotoSize'),
   Audio: require('./Audio'),
   Document: require('./Document'),
   Sticker: require('./Sticker'),
   Video: require('./Video'),
   Voice: require('./Voice'),
   Contact: require('./Contact'),
   Location: require('./Location'),
   Venue: require('./Venue'),
   UserProfilePhotos: require('./UserProfilePhotos'),
   File: require('./File'),
   ReplyKeyboardMarkup: require('./ReplyKeyboardMarkup'),
   KeyboardButton: require('./KeyboardButton'),
   ReplyKeyboardHide: require('./ReplyKeyboardHide'),
   InlineKeyboardMarkup: require('./InlineKeyboardMarkup'),
   InlineKeyboardButton: require('./InlineKeyboardButton'),
   CallbackQuery: require('./CallbackQuery'),
   ForceReply: require('./ForceReply'),
   ChatMember: require('./ChatMember'),
   InlineQuery: require('./InlineQuery'),
   InlineQueryResultArticle: require('./InlineQueryResultArticle'),
   InlineQueryResultPhoto: require('./InlineQueryResultPhoto'),
   InlineQueryResultGif: require('./InlineQueryResultGif'),
   InlineQueryResultMpeg4Gif: require('./InlineQueryResultMpeg4Gif'),
   InlineQueryResultVideo: require('./InlineQueryResultVideo'),
   InlineQueryResultAudio: require('./InlineQueryResultAudio'),
   InlineQueryResultVoice: require('./InlineQueryResultVoice'),
   InlineQueryResultDocument: require('./InlineQueryResultDocument'),
   InlineQueryResultLocation: require('./InlineQueryResultLocation'),
   InlineQueryResultVenue: require('./InlineQueryResultVenue'),
   InlineQueryResultContact: require('./InlineQueryResultContact'),
   InlineQueryResultCachedPhoto: require('./InlineQueryResultCachedPhoto'),
   InlineQueryResultCachedGif: require('./InlineQueryResultCachedGif'),
   InlineQueryResultCachedMpeg4Gif: require('./InlineQueryResultCachedMpeg4Gif'),
   InlineQueryResultCachedSticker: require('./InlineQueryResultCachedSticker'),
   InlineQueryResultCachedDocument: require('./InlineQueryResultCachedDocument'),
   InlineQueryResultCachedVideo: require('./InlineQueryResultCachedVideo'),
   InlineQueryResultCachedVoice: require('./InlineQueryResultCachedVoice'),
   InlineQueryResultCachedAudio: require('./InlineQueryResultCachedAudio'),
   InputTextMessageContent: require('./InputTextMessageContent'),
   InputLocationMessageContent: require('./InputLocationMessageContent'),
   InputVenueMessageContent: require('./InputVenueMessageContent'),
   InputContactMessageContent: require('./InputContactMessageContent'),
   ChosenInlineResult: require('./ChosenInlineResult'),
   InputMessageContent: require('./InputMessageContent'),
   InlineQueryResult: require('./InlineQueryResult'),
}
